www/js/ble2mqtt.js

```diff
@@ -1,137 +1,298 @@
+    document.getElementById('message').innerText = message;
+    setTimeout(() => document.getElementById('message').style.display = 'none', 3000);
 }
 
-/* OTA */
-function otaStartUpload(type) {
-  var file = $('#' + type + '-file')[0].files[0];
-  if (!file)
-    return;
-
-  $('[id$=upload]').attr('disabled', true);
-  $('[id$=file]').attr('disabled', true);
-  $('#ota-spinner').show();
-  $('#ota-status').text('Uploading');
-
-  $.ajax({
-    url: '/ota/' + type,
-    type: 'POST',
-    data: file,
-    processData: false,
-  })
-  .done(onOtaUploadCompleted)
-  .fail(onOtaUploadCompleted);
+function restart() {
+    fetch(RESTART, {
+        method: 'POST',
+    })
+        .then((res) => {
+            console.log("restart")
+            window.open("/busy.html", "_self")
+        })
+        .catch((err) => {
+            document.getElementById('message').innerText = "Can't restart";
+            console.error(err)
+        })
 }
 
-function onOtaUploadCompleted(data, textStatus, jqXHR) {
-  $('#ota-status').text(textStatus);
-  $('#ota-spinner').hide();
-  $('[id$=file]').attr('disabled', false);
-  $('[id$=upload]').attr('disabled', false);
+function getStatus() {
+    fetch(STATUS)
+        .then(response => response.json())
+        .then(json => {
+            document.getElementById('software-version').innerHTML = json.version;
+        });
 }
 
-/* File Manager */
-function fileManagerGetFilesList() {
-  path = $('#file-manager-path').text();
-
-  $('#file-manager-list').empty();
-  $.ajax({
-    url: '/fs' + path,
-    type: 'GET',
-    processData: false,
-  })
-  .done(onFileManagerFilesReady);
+function getLatestReleaseInfo() {
+    fetch('https://api.github.com/repos/shmuelzon/esp32-ble2mqtt/tags')
+        .then(response => response.json())
+        .then(json => {
+            let release = json[0];
+            if (release) document.getElementById('latest-release-id').innerHTML = `(latest <a href="${LATEST_RELEASE}" target="_blank">${release.name}</a>)`;
+        });
 }
 
-function onFileManagerFilesReady(data, textStatus, jqXHR) {
-  for (entry of data) {
-    var div = $('<div class="file-manager-entry file-manager-entry-' + entry.type + '"></div>')
-      .click(onfileManagerClick)
-      .dblclick(entry.type, onfileManagerDoubleClick);
-    div.append($('<span class="file-manager-name"></span').text(entry.name));
-    if (entry.type === 'file')
-      div.append($('<span class="file-manager-size"></span').text(entry.size));
-    $('#file-manager-list').append(div);
-  }
+function otaStartUpload(type) {
+    let file = document.getElementById(type + '-file').files[0];
+    if (!file)
+        return;
+    progress(true);
+    fetch(OTA_UPLOAD + type, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/octet-stream'
+        },
+        body: file
+    })
+        .then((res) => {
+            document.getElementById('message').innerText = "Upload complete";
```

Comment:
Shouldn't this be a call to `toaster()`?

Reply:
yep